originally, the gettid wrapper was needed on glibc and musl,
glibc-2.30 added a gettid function, so now it is only needed on musl

--- tools/profiler/core/platform.h.orig
+++ tools/profiler/core/platform.h
@@ -56,7 +56,7 @@

 // We need a definition of gettid(), but glibc doesn't provide a
 // wrapper for it.
-#if defined(__GLIBC__)
+#if defined(__linux__) && !defined(__GLIBC__)
 #include <unistd.h>
 #include <sys/syscall.h>
 static inline pid_t gettid()
